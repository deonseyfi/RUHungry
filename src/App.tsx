import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import logo from './logo.svg';
import './App.css';

type User = {
    name: string;
    age: number;
};

const App: React.FC = () => {
    const [users, setUsers] = useState<User[]>();

    const getUsers = () => {
        fetch('/api/users')
            .then((res) => res.json())
            .then(setUsers);
    };

    return (
        <div className='App'>
            <header className='App-header'>
                <img src={logo} className='App-logo' alt='logo' />
                <p>
                    Edit <code>src/App.tsx</code> and save to reload.
                </p>
                <a className='App-link' href='https://reactjs.org'>
                    Learn React
                </a>
                <br />
                <Button variant='contained' color='primary' startIcon={<AccountCircleIcon />} onClick={getUsers}>
                    Fetch users in DB
                </Button>
                {users && (
                    <ul>
                        {users.map((user) => (
                            <li key={user.name}>{`${user.name}, ${user.age} years old`}</li>
                        ))}
                    </ul>
                )}
            </header>
        </div>
    );
};

export default App;
