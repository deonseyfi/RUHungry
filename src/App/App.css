.App {
    text-align: center;
    background-color: #718cc2;
    overflow: hidden;
}

.App-logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .App-logo {
        animation: App-logo-spin infinite 20s linear;
    }
}
li {
    list-style-type: none;
    max-width: 60%;
  }
.card-details-photos{
    max-height: 17.3rem;
    max-width: 17.3rem;
    height: auto;
    width: auto;
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 1rem;
    background-size: contain;
    background-repeat: no-repeat;
    background-attachment: fixed;
}

.App-header {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.App-link {
    color: #61dafb;
    background: none;
    border: none;
    font-size: calc(10px + 2vmin);
    text-decoration: underline;
    cursor: pointer;
    outline: none;
}
.Card-list {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    text-align: start;
}
.Card-Details{
    text-align: center;
    display: 'flex';
    flex-direction: column;
    justify-content: center;
    align-self: flex-start;
}
.card-details-header{
    display: flex;
    flex-direction: column;
}
.yelp-logo{
    display: flex;
    flex-direction: column;
    align-items: center;
}
.yelp-logo-text{
    font-size: x-small;
    color: #a5a5a5;
    padding-bottom: 12px;
    margin-top: -8px;
    display: flex;
    justify-content: flex-start;
    margin-left: 1.37rem;
}
.yelp-ratings{
    display: flex;
    flex-direction: row;
    align-items: center;
}
.yelp-review-count{
    color: #a5a5a5;
    display: flex;
    align-self: center;
    margin-left: -6px;
    margin-top: 2px;
}
.yelp-contents{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding-top: 16px;
}
.card-details-buttons{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    padding-top: 32px;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
