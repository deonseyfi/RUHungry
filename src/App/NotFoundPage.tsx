import { Button } from '@material-ui/core';
import React from 'react';
import { Link } from 'react-router-dom';

/**
 * 404 page to hit if directed to invalid path.
 *
 * @returns {React.ReactElement} Not Found Page component.
 */
const NotFoundPage = (): React.ReactElement => (
    <div style={{ textAlign: 'center' }}>
        <h1>404 NOT FOUND</h1>
        <Button
            variant='outlined'
            color='inherit'
            component={Link}
            to='/'
        >Go Home
        </Button>
    </div>
);
export default NotFoundPage;
