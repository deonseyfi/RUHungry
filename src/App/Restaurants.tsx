import { Button } from '@material-ui/core';
import React from 'react';
import { Link, useParams } from 'react-router-dom';
import CardDetails from '../components/CardDetails';
import { Restaurant, RestaurantDetails } from './types';

// TODO: Food cards will go here.

/**
 * Takes in url string and splits it into individual categories.
 *
 * @param {string} list Single string of the pulled category url.
 * @param {number} index Index of the split list array.
 * @param catList
 * @returns {string} Category name at specific index.
 */
const getCategory = (catList: string): string => {
    const categorySize = catList.includes('&') ? 2 : 1;
    const cat1 =
        categorySize > 1
            ? catList.substring(catList.indexOf('=') + 1, catList.indexOf('&'))
            : catList.substring(catList.indexOf('=') + 1);
    if (categorySize > 1) {
        const cat2 = catList.substring(catList.lastIndexOf('=') + 1);
        return `${cat1} and ${cat2}`;
    }
    return `${cat1}`;
};

/**
 * Restaurants results page with card stack and chosen categories.
 *
 * @returns {React.ReactElement} Restaurants page component.
 */
const Restaurants = (): React.ReactElement => {
    // TODO: Terrible spaghetti, research a better way to split url output. (For testing only)
    const { categories } = useParams();
    const catList = categories !== undefined ? categories : 'null';
    const categoryText = getCategory(catList);

    const [open, setOpen] = React.useState(false);
    /**
     * Function to open or close the Dialog.
     */
    const handleOpenClose = () => {
        setOpen(!open);
    };
    const restaurant: Restaurant = {
        // Pull info from API, this is for testing purposes only.
        // https://api.yelp.com/v3/businesses/search?{params}
        // eslint-disable-next-line spellcheck/spell-checker
        id: 'krs94qCUqsxEey1rFXyAhg',
        name: "Dave's Hot Chicken",
        price: 2,
        rating: 4,
        reviewCount: 320,
        categories: [
            { alias: 'chickenshop' },
            { alias: 'american' },
        ],
        imageUrl: 'https://s3-media3.fl.yelpcdn.com/bphoto/d08yu8NH0qiIUG5PkGPdHA/o.jpg',
    };
    const restaurantDetail: RestaurantDetails = {
        // Pull info from API, this is for testing purposes only.
        // https://api.yelp.com/v3/businesses/{id}
        displayAddress: '1350 E Colorado St, Glendale, CA 91205',
        displayPhone: '(818) 937-9488',
        isOpenNow: true,
        // eslint-disable-next-line spellcheck/spell-checker
        // eslint-disable-next-line max-len
        url: 'https://www.yelp.com/biz/daves-hot-chicken-glendale-3?adjust_creative=Rit42U9ivRMCkc_nRd-X1A&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_lookup&utm_source=Rit42U9ivRMCkc_nRd-X1A',
        photos: [
            'https://s3-media3.fl.yelpcdn.com/bphoto/d08yu8NH0qiIUG5PkGPdHA/o.jpg',
            'https://s3-media2.fl.yelpcdn.com/bphoto/AwXNZ9owcpwvDYwamslGTg/o.jpg',
            'https://s3-media1.fl.yelpcdn.com/bphoto/YFKwJTfpfmkYkzYcn5mwrg/o.jpg',
        ],
    };

    /**
     * Function to return a formatted map link of the restaurant.
     *
     * @param {string} name The name of the restaurant.
     * @param {string} address The full address of the restaurant.
     * @returns {string} Map link of the selected restaurant.
     */
    const getLink = (name: string, address: string): string =>
        `https://maps.apple.com/?address=${encodeURIComponent(address)}&q=${encodeURIComponent(name)}`;

    /**
     * Returns the location of the appropriate rating image (Using this method to avoid importing all images).
     *
     * @param {number} rating The restaurant's rating.
     * @returns {string} The location of the appropriate rating image.
     */
    const getRating = (rating: number): string => `${process.env.PUBLIC_URL}/yelp-stars/${rating}.png`;
    return (
        <div>
            <Button
                variant='outlined'
                color='inherit'
                component={Link}
                to={{ pathname: '/home' }}
            >
                New Search
            </Button>
            <h1>(DEV) You have selected {categoryText}</h1>
            <Button variant='outlined' color='inherit' onClick={handleOpenClose}>
                Click To Open details of {restaurant.name}
            </Button>
            <CardDetails
                open={open}
                onClose={handleOpenClose}
                restaurantTitle={restaurant.name}
                mapLink={getLink(restaurant.name, restaurantDetail.displayAddress)}
                address={restaurantDetail.displayAddress}
                phoneNumber={restaurantDetail.displayPhone}
                photos={restaurantDetail.photos[0]}
                yelpUrl={restaurantDetail.url}
                ratingImage={getRating(restaurant.rating)}
                reviewCount={restaurant.reviewCount}
            />
        </div>
    );
};
export default Restaurants;

/*
    This works the same as above, just takes 2 separate slugs instead of parsing one.
    Pro is easier to read, con is there needs to be an extra route for only one category in App.
    Can also just remove the first=x and second=y in the path to avoid too much splitting.
    Ex: restaurants/x&y
    <Route path='/restaurants/:cat1&:cat2' element={<Restaurants />} />
    <Route path='/restaurants/:cat1' element={<Restaurants />} />
*/

/*
const Restaurants = (): React.ReactElement => {
    const { cat1, cat2 } = useParams();
    const one = cat1?.split('&')[0].split('=')[1];
    const two = cat2?.split('&')[0].split('=')[1];
    return (
        <div>
            {cat2 !== undefined && (<h1>{one} and {two}</h1>)}
            {cat2 === undefined && (<h1>{one}</h1>)}
        </div>
    );
};
export default Restaurants;
*/
