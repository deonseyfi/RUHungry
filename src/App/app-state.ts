/**
 * States used for App.
 */
interface AppState {
    /**
     * Value of the first chosen food category.
     */
    firstCategoryValue: string;
    /**
     * Value of the second chosen food category.
     */
    secondCategoryValue: string;
    /**
     * Boolean to check if second category is desired.
     */
    addCategory: boolean;
}
export default AppState;
