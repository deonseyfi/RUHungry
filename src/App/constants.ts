/* eslint-disable spellcheck/spell-checker */
import { FoodCategoryList } from './types';

// Complete list of food categories.
const CATEGORIES: FoodCategoryList = [
    { key: '', label: '', symbol: '' },
    { key: 'all', label: 'All', symbol: 'All' },
    { key: 'burgers', label: 'Burgers', symbol: '🍔' },
    { key: 'hotdogs', label: 'Hot Dogs', symbol: '🌭' },
    { key: 'mexican', label: 'Mexican', symbol: '🌮' },
    { key: 'pizza', label: 'Pizza', symbol: '🍕' },
    { key: 'italian', label: 'Italian', symbol: '🍝' },
];
export default CATEGORIES;

// // Complete list of food categories.
// const CATEGORIES: CategoryList = [
//     { category: { value: '', symbol: '' } },
//     { category: { value: 'All', symbol: 'All' } },
//     { category: { value: 'Burgers', symbol: '🍔' } },
//     { category: { value: 'Hot Dogs', symbol: '🌭' } },
//     { category: { value: 'Mexican ', symbol: '🌮' } },
//     { category: { value: 'Pizza', symbol: '🍕' } },
//     { category: { value: 'Italian ', symbol: '🍝' } },
// ];
// export default CATEGORIES;
