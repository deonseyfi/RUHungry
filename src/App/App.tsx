import React from 'react';
import Button from '@material-ui/core/Button';
import { IconButton } from '@material-ui/core';
import HighlightOffSharpIcon from '@material-ui/icons/HighlightOffSharp';
import './App.css';
import AppState from './app-state';
import Select from '../components/Select';
import StartSplash from '../components/StartSplash';
import LuckyButton from '../components/LuckyButton';
import CATEGORIES from './constants';
import { FoodCategoryList } from './types';

/**
 * Main App class.
 */
class App extends React.Component<{}, AppState> {
    /**
     * App constructor.
     */
    public constructor(props: {}) {
        super(props);
        this.state = {
            clickedStart: false,
            firstCategoryValue: 'All',
            secondCategoryValue: '',
            addCategory: false,
        };
        this.handleClickedStart = this.handleClickedStart.bind(this);
        this.handleFirstCategoryChange = this.handleFirstCategoryChange.bind(this);
        this.handleSecondCategoryChange = this.handleSecondCategoryChange.bind(this);
        this.handleNextButton = this.handleNextButton.bind(this);
    }

    /**
     * Move from Home to Cravings.
     */
    public handleClickedStart = () => {
        this.setState({ clickedStart: true });
    };

    /**
     * Function to change the current selected option in the first category.
     *
     * @param {string} value The first Select's value.
     */
    public handleFirstCategoryChange = (value: string) => {
        this.setState({ firstCategoryValue: value });
    };

    /**
     * Function to change the current selected option in the second category.
     *
     * @param {string} value The second Select's value.
     */
    public handleSecondCategoryChange = (value: string) => {
        this.setState({ secondCategoryValue: value });
    };

    /**
     * Temporarily alert which food categories have been selected.
     */
    public handleNextButton = () => {
        if (this.state.addCategory) {
            alert(`Chosen ${this.state.firstCategoryValue} and ${this.state.secondCategoryValue}`);
        } else {
            alert(`Chosen ${this.state.firstCategoryValue}`);
        }
    };

    /**
     * Renders App.
     *
     * @returns {React.ReactNode} Rendered App.
     */
    render(): React.ReactNode {
        // List of Categories for first Select.
        // If second category, remove All, blank, and second category value.
        // Else, remove blank.
        const firstCategoryList: FoodCategoryList = CATEGORIES.filter((category) =>
            (this.state.addCategory
                ? category.value !== this.state.secondCategoryValue && category.value !== '' && category.value !== 'All'
                : category.value !== ''),
        );

        // New List of Categories for second Select.
        // If second category value is blank, remove All and first category value.
        // Else, remove All, first category value, and blank.
        const secondCategoryList: FoodCategoryList = CATEGORIES.filter((category) =>
            (this.state.secondCategoryValue === ''
                ? category.value !== this.state.firstCategoryValue && category.value !== 'All'
                : category.value !== this.state.firstCategoryValue && category.value !== 'All' && category.value !== ''),
        );
        return (
            <div className='App'>
                <div className='App-header'>
                    {!this.state.clickedStart && <StartSplash onClick={this.handleClickedStart} />}
                    {this.state.clickedStart && (
                        <>
                            <Select
                                value={this.state.firstCategoryValue}
                                onChange={(event): void => {
                                    this.handleFirstCategoryChange(String(event.target.value));
                                }}
                            >
                                {firstCategoryList}
                            </Select>
                            <div>(DEV) The current selected value is {this.state.firstCategoryValue}</div>
                            <br />
                            {this.state.firstCategoryValue !== 'All' && this.state.addCategory === false && (
                                <Button
                                    id='add'
                                    variant='outlined'
                                    color='inherit'
                                    onClick={() => {
                                        this.setState({ addCategory: true });
                                    }}
                                >
                                    {' '}
                                    Add Another Category
                                </Button>
                            )}
                            {this.state.addCategory === true && (
                                <>
                                    <div
                                        style={{
                                            display: 'flex',
                                            flexDirection: 'row',
                                            justifyContent: 'center',
                                            paddingLeft: '3rem',
                                        }}
                                    >
                                        <Select
                                            value={this.state.secondCategoryValue}
                                            onChange={(event): void => {
                                                this.handleSecondCategoryChange(String(event.target.value));
                                            }}
                                        >
                                            {secondCategoryList}
                                        </Select>
                                        <IconButton
                                            size='medium'
                                            onClick={() => {
                                                this.setState({ addCategory: false });
                                                this.setState({ secondCategoryValue: '' });
                                            }}
                                        >
                                            <HighlightOffSharpIcon color='secondary' />
                                        </IconButton>
                                    </div>
                                    <div>
                                        The current selected value is{' '}
                                        {this.state.secondCategoryValue === ''
                                            ? 'Blank'
                                            : this.state.secondCategoryValue}
                                    </div>
                                </>
                            )}
                            <br />
                            <br />
                            <br />
                            {(!this.state.addCategory ||
                                (this.state.addCategory && this.state.secondCategoryValue !== '')) && (
                                <Button variant='outlined' color='inherit' onClick={this.handleNextButton}>
                                    Next
                                </Button>
                            )}
                            <br />
                            <br />
                            <br />
                            <LuckyButton />
                        </>
                    )}
                </div>
            </div>
        );
    }
}
export default App;
