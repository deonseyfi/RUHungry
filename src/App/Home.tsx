import React from 'react';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import HighlightOffSharpIcon from '@material-ui/icons/HighlightOffSharp';
import { Link } from 'react-router-dom';
import './App.css';
import AppState from './app-state';
import Select from '../components/Select';
import StartSplash from '../components/StartSplash';
import LuckyButton from '../components/LuckyButton';
import CATEGORIES from './constants';
import { FoodCategoryList } from './types';

/**
 * RUHungry Home Page.
 */
class Home extends React.Component<{}, AppState> {
    /**
     * Home constructor.
     */
    public constructor(props: {}) {
        super(props);
        this.state = {
            clickedStart: false,
            firstCategoryValue: 'All',
            secondCategoryValue: '',
            addCategory: false,
        };
        this.handleClickedStart = this.handleClickedStart.bind(this);
        this.handleFirstCategoryChange = this.handleFirstCategoryChange.bind(this);
        this.handleSecondCategoryChange = this.handleSecondCategoryChange.bind(this);
    }

    /**
     * Move from Home to Cravings.
     */
    public handleClickedStart = () => {
        this.setState({ clickedStart: true });
    };

    /**
     * Function to change the current selected option in the first category.
     *
     * @param {string} value The first Select's value.
     */
    public handleFirstCategoryChange = (value: string) => {
        this.setState({ firstCategoryValue: value });
    };

    /**
     * Function to change the current selected option in the second category.
     *
     * @param {string} value The second Select's value.
     */
    public handleSecondCategoryChange = (value: string) => {
        this.setState({ secondCategoryValue: value });
    };

    /**
     * Clear all selected categories to original state.
     */
    public handleClearButton = () => {
        this.setState({ firstCategoryValue: 'All' });
        this.setState({ secondCategoryValue: '' });
        this.setState({ addCategory: false });
    };

    /**
     * Renders Home Page.
     *
     * @returns {React.ReactNode} Rendered Home Page.
     */
    render(): React.ReactNode {
        // List of Categories for first Select.
        // If second category, remove All, blank, and second category value.
        // Else, remove blank.
        const firstCategoryList: FoodCategoryList = CATEGORIES.filter((category) =>
            (this.state.addCategory
                ? category.label !== this.state.secondCategoryValue && category.label !== '' && category.label !== 'All'
                : category.label !== ''),
        );

        // New List of Categories for second Select.
        // If second category value is blank, remove All and first category value.
        // Else, remove All, first category value, and blank.
        const secondCategoryList: FoodCategoryList = CATEGORIES.filter((category) =>
            (this.state.secondCategoryValue === ''
                ? category.label !== this.state.firstCategoryValue && category.label !== 'All'
                : category.label !== this.state.firstCategoryValue && category.label !== 'All' && category.label !== ''),
        );
        /**
         * Returns the path to navigate to when Next button is clicked.
         *
         * @returns {string} The path to navigate to.
         */
        const getRestaurantsPath = (): string =>
            (this.state.secondCategoryValue !== ''
                ? `/restaurants/first=${this.state.firstCategoryValue}&second=${this.state.secondCategoryValue}`
                : `/restaurants/first=${this.state.firstCategoryValue}`);
        return (
            <>
                {!this.state.clickedStart && <StartSplash onClick={this.handleClickedStart} />}
                {this.state.clickedStart && (
                    <>
                        <div
                            style={{
                                display: 'flex',
                                flexDirection: 'row',
                                justifyContent: 'right',
                                paddingLeft: '18rem',
                            }}
                        >
                            {this.state.firstCategoryValue !== 'All' && (
                                <Button
                                    id='clear'
                                    variant='outlined'
                                    color='secondary'
                                    onClick={() => {
                                        this.handleClearButton();
                                    }}
                                >
                                    Clear
                                </Button>
                            )}
                        </div>
                        <Select
                            value={this.state.firstCategoryValue}
                            onChange={(event): void => {
                                this.handleFirstCategoryChange(String(event.target.value));
                            }}
                        >
                            {firstCategoryList}
                        </Select>
                        <br />
                        {this.state.firstCategoryValue !== 'All' && this.state.addCategory === false && (
                            <Button
                                id='add'
                                variant='outlined'
                                color='inherit'
                                onClick={() => {
                                    this.setState({ addCategory: true });
                                }}
                            >
                                Add Another Category
                            </Button>
                        )}
                        {this.state.addCategory === true && (
                            <>
                                <div
                                    style={{
                                        display: 'flex',
                                        flexDirection: 'row',
                                        justifyContent: 'center',
                                        paddingLeft: '3rem',
                                    }}
                                >
                                    <Select
                                        value={this.state.secondCategoryValue}
                                        onChange={(event): void => {
                                            this.handleSecondCategoryChange(String(event.target.value));
                                        }}
                                    >
                                        {secondCategoryList}
                                    </Select>
                                    <IconButton
                                        size='medium'
                                        onClick={() => {
                                            this.setState({ addCategory: false });
                                            this.setState({ secondCategoryValue: '' });
                                        }}
                                    >
                                        <HighlightOffSharpIcon color='secondary' />
                                    </IconButton>
                                </div>
                            </>
                        )}
                        <br />
                        <br />
                        <br />
                        {(!this.state.addCategory ||
                            (this.state.addCategory && this.state.secondCategoryValue !== '')) && (
                            <Button variant='outlined' color='inherit' component={Link} to={getRestaurantsPath()}>
                                Next
                            </Button>
                        )}
                        <br />
                        <br />
                        <br />
                        <LuckyButton />
                    </>
                )}
            </>
        );
    }
}
export default Home;
