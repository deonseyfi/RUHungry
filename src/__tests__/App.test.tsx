import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App/App';

configure({ adapter: new Adapter() });

describe('App test', () => {
    test('renders initial App', () => {
        const wrapper = mount(<App />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
