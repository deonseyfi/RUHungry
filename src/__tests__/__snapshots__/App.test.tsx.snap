// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test renders initial App 1`] = `ReactWrapper {}`;
